modules/ensemble/lib/framework/apiproviders/firestore/firestore_app.dart

```diff
@@ -3,7 +3,7 @@ import 'package:cloud_firestore/cloud_firestore.dart';
 class FirestoreApp {
   final FirebaseFirestore firestore;
   FirestoreApp(this.firestore);
-  Query getQuery(Map api, {bool isCollectionGroup = false}) {
+  Future<dynamic> getQuery(Map api, {bool isCollectionGroup = false}) async {
```

Comment:
@TheNoumanDev getQuery is also called inside subscribeToApi function and that expects a Query object back. Does your code even compile anymore? did you even test the firestore api provider?